os/include/tinyara/input/ist415.h

```diff
@@ -0,0 +1,315 @@
+#define HCOM_SET_GRIP_DATA				0xB1
+#define HCOM_SET_POCKET_MODE			0xB2
+#define HCOM_SET_CALL_MODE				0xB3
+#define HCOM_SET_HALF_AOD_MODE			0xB4
+
+#define HCOM_GET_SELFTEST_DATA			0xC0
+#define HCOM_GET_SELFTEST_RES			0xC1
+#define HCOM_GET_TOUCH_EN				0xC2
+#define HCOM_GET_WET_STATUS				0xC3
+#define HCOM_GET_FOD_INFO				0xC4
+#define HCOM_GET_FOD_POS				0xC5
+
+// USE HID Only
+#define HCOM_SET_USE_LIBCLOSE			0xD0
+#define HCOM_GET_DEVICE_DESC			0xD1
+#define HCOM_REPORT_COORD				0xD2
+#define HCOM_GET_REPORT_DESC			0xD3
+#define HCOM_CMD_REG					0xD4
+#define HCOM_DAT_REG					0xD5
+#define HCOM_INPUT_REPORT_REG			0xD6
+#define HCOM_OUTPUT_REPORT_REG			0xD7
+#define	HCOM_USE_REPORT_COORD			0xD8
+
+#define HCOM_SET_REC_MODE				0xF0
+#define HCOM_SET_REG					0xF1
+#define HCOM_GENERATE_INT				0xF2
+#define HCOM_GET_DOMEKEY_STATUS			0xF3
+
+#define HCOM_SW_DA						0xFE
+
+#define EVENT_BYTE						16
+
+// EventId (stEventCoord_t)
+#define EID_MASK						0x3
+#define EID_COORD						0
+#define EID_STATUS						1
+#define EID_GESTURE						2
+
+#define LEFT_NUM_MASK					0x3F
+
+// TouchStatus
+#define TCH_PRESS						2
+
+// StatusType (stEventStatus_t)
+#define STS_TYPE_CMD_DRIVEN				0
+#define STS_TYPE_VENDOR_INFO			7
+
+// StatusId (stEventStatus_t)
+#define STS_ID_CMD_CALIB				0
+#define STS_ID_VENDOR_SELFTEST			0
+
+// aStatusData (stEventStatus_t)
+#define STS_DATA_FW_INTEGRITY_PASS		0x80
+#define STS_DATA_FW_INTEGRITY_FAIL		0x40
+#define STS_DATA_CAL_PASS				0x10
+#define STS_DATA_CAL_FAIL				0x80
+#define STS_DATA_CMCS_PASS				0x10
+#define STS_DATA_CMCS_FAIL				0x80
+#define STS_DATA_CMCS_CM				0x00
+#define STS_DATA_CMCS_CMJIT				0x02
+
+#define PARSE_HW_ID(n)					((n >> 10) & 0x3)
+#define HW_ID							0x01
+
+#define PARSE_TACT_SWITCH(n)			((n >> 3) & 0x1)
+
+#define SELFTEST_CM						(1 << 0)
+#define SELFTEST_JITTER					(1 << 2)
+
+#define SELFTEST_CM_MSG					(0xC3)
+#define SELFTEST_JITTER_MSG				(0x71)
+
+#define SELFTEST_PASS					(0x10)
+#define SELFTEST_FAIL					(0x80)
+
+#define MTL_CR_REGISTER					0x40000000
+#define SLF_CR_REGISTER					0x40001768
+
+#define HW_ID_REGISTER					0x30030000
+
+#pragma pack(push,1)
+typedef struct {
+	u8	ButtonState			:2;
+	u8	FingerID			:3;
+	u8	Reserved			:3;
+	u16 X;
+	u16 Y;
+} stEventCoordHID_t;
+
+typedef struct {
+	u16 ReportLength;
+	u8  ReportID;
+
+	stEventCoordHID_t stCoord[MAX_FINGER_NUM];
+	
+	u16 Scantime;
+	u8  ContactCount;
+	u8  Button;
+} stEventReportHID_t;
+#pragma pack(pop)
+
+typedef struct {
+	u8 EventId		:2;
+	u8 StatusType	:4;
+	u8 SFOn			:2;
+	u8 StatusId;
+	u8 aStatusData[5];
+	u8 LeftEvt		:5;
+	u8 Reserved		:3;
+	u8 Reserved0;
+	u8 Reserved1;
+	u8 Reserved2;
+	u8 Reserved3;
+	u8 Reserved4;
+	u8 Reserved5;
+	u8 Reserved6;
+	u8 Reserved7;
+} stEventStatus_t;
+
+typedef struct{
+	u32 ChipId;
+
+	u32 ReleaseVer;
+	
+	u32 LcdWidth;
+	u32 LcdHeight;
+	
+	u32 NumOfRx;
+	u32 NumOfTx;
+
+	u16 Crc;
+
+	u8 SwapInfo;
+	
+	u32 CalResult;
+	u32 SCalResult;
+}stIST415xInfo_t;

```

Comment:
let's add space.
And this struct is necessary?

Reply:
we have taken this from tizen applied code. yes, It's not necessary, so I have removed it.